src/tests/unit_tls_policy.cpp

```diff
@@ -140,15 +140,27 @@ class TLS_Policy_Unit_Tests final : public Test {
          Botan::TLS::Policy default_policy;
          result.test_eq(
             "default TLS Policy offers exactly one", default_policy.key_exchange_groups_to_offer().size(), 1);
-         result.confirm(
-            "default TLS Policy offers preferred group",
-            default_policy.key_exchange_groups().front() == default_policy.key_exchange_groups_to_offer().front());
+
+         auto first_pure_ecc_group =
+            [](const std::vector<Botan::TLS::Named_Group>& groups) -> std::optional<Botan::TLS::Named_Group> {
```

Comment:
Nit: `std::span<>` would work here as well and erase the `std::vector<>` type, if we were to return any other type from `Policy::key_exchange_groups()` for some reason in the future.

```suggestion
            [](std::span<const Botan::TLS::Named_Group> groups) -> std::optional<Botan::TLS::Named_Group> {
```